core/frontend/src/tile/map/MapTileTree.ts

```diff
@@ -414,6 +449,86 @@ export class MapTileTree extends RealityTileTree {
     }
   }
 
+  /** @internal */
+  public override reportTileVisibility(args: TileDrawArgs, selected: RealityTile[]) {
+
+    const debugControl = args.context.target.debugControl;
+
+    const layersVisibilityBefore =  this.cloneImageryTreeState();
+
+    const changes =  new Array<MapLayerScaleRangeVisibility> ();
+
+    if (!layersVisibilityBefore)
+      return;
+
+    for (const [treeId] of layersVisibilityBefore) {
+      const treeVisibility = this.getImageryTreeState(treeId);
+      if (treeVisibility) {
+        treeVisibility.reset();
+      }
+    }
+
+    const out: string[] = [];
+    const vis: string[] = [];
+
+    let allTilesRead = true;
+    for (const selectedTile of selected) {
+      if (selectedTile instanceof MapTile) {
+        if (!selectedTile.isReady)
+          allTilesRead = false;
+        let selectedImageryTiles = selectedTile.imageryTiles;
+        if (selectedTile.hiddenImageryTiles) {
+          selectedImageryTiles = selectedImageryTiles ? [...selectedImageryTiles, ...selectedTile.hiddenImageryTiles] : selectedTile.hiddenImageryTiles;
+        }
+        if (selectedImageryTiles) {
+          for (const selectedImageryTile of selectedImageryTiles) {
+            const treeState = this.getImageryTreeState(selectedImageryTile.tree.id);
+            if (treeState) {
+              if (selectedImageryTile.isOutOfLodRange) {
+                out.push(selectedImageryTile.contentId);
+                treeState.setScaleRangeVisibility(false);
+              } else {
+                treeState.setScaleRangeVisibility(true);
+                vis.push(selectedImageryTile.contentId);
+              }
+            }
+
+          }
+        }
+      }
+    }
+
+    if (!allTilesRead)
+      changes;
+
+    for (const [treeId, prevState] of layersVisibilityBefore) {

```

Comment:
@pmconne While this report function get the job done, I think that would be more efficient if selected tiles would retain information about the original imageryTree index in the MapTileTree.. that way I would know which map-layer is corresponding the a given selected tile.  Problem is, we are passing RealityTile objects, and it makes no sense to keep maplayer related information on that object since a tile can be shared by two different map-layers...  some sort of wrapper 'RealityTileSection' object would be needed but that would impact several API.  What do you think?

Reply:
My initial reaction is to prefer to keep this concept from propagating any further than necessary through the rest of the API.

Reply:
@pmconne So are you good with actual design?  This has been dragging for a while now...

Reply:
> MapTile API already suffers from a dearth of documentation and an abundance of complexity. Please document any new things you add to it (including @internal things) explaining to our future selves what they do and why.

If you document what you added, I will have a better understanding of the design.
I have questions but I was hoping you would describe your design instead of requiring the reader to infer it.

Reply:
Your api md diffs still contain undocumented APIs. Documentation is **required** for beta and public APIs. It is strongly encouraged for all APIs.

Reply:
You still have undocumented beta APIs.